src/lib/eliom_reference.server.ml

```diff
@@ -176,20 +176,20 @@ let eref_from_fun_ ~ext ~scope ?secure ?persistent f : 'a eref =
     | `Global ->
       begin
         match persistent with
-          | None -> (Volatile.eref_from_fun_ ~ext ~scope ?secure f :> _ eref)
-          | Some name ->
+          | Some name when not @@ Eliom_common.Persistent_tables.check_if_omitting_storage () ->
             (f, ext,
              Ocsiper (
                pers_ref_store >>= fun store ->
                Ocsipersist.make_persistent ~store ~name ~default:None))
+          | _ -> (Volatile.eref_from_fun_ ~ext ~scope ?secure f :> _ eref)
       end
     | `Site ->
       begin
         match persistent with
-          | None -> (Volatile.eref_from_fun_ ~ext ~scope ?secure f :> _ eref)
-          | Some name ->
+          | Some name when not @@ Eliom_common.Persistent_tables.check_if_omitting_storage () ->
 (*VVV!!! ??? CHECK! *)
               (f, ext, Ocsiper_sit (Ocsipersist.open_table name))
+          | _ -> (Volatile.eref_from_fun_ ~ext ~scope ?secure f :> _ eref)
```

Comment:
I'm not sure about this, since references are usually created at toplevel, in a context where `Eliom_common.Persistent_tables.check_if_omitting_storage ()` is false. For instance:
```
let%server has_followed =
  Eliom_reference.eref ~persistent:"Bs_ref__has_followed"
    ~scope:Eliom_common.default_group_scope false
```

Reply:
In this specific case this is not a problem, since for `default_group_scope` the data is stored lazily :
```
  let eref_from_fun_ ~ext ~scope ?secure f : 'a eref =
    f, ext, match scope with
      | `Request -> Req (Polytables.make_key ())
      | `Global -> Ref (ref (Lazy.from_fun f))
      | `Site -> Sit (Polytables.make_key ())
      | (#Eliom_common.user_scope as scope) ->
        Vol (lazy (create_volatile_table ~scope ?secure ()))
```

For the more general case I'm wondering whether the situation you're mentioning is even possible. Will Eliom ever store persistent data associated with some cookie C from some context where C is not known? Investigating...

Reply:
Why are you quoting `Volatile.eref_from_fun_`? Aren't we going to line 199?
```ocaml
          (f, ext, Per (create_persistent_table ~scope ?secure name))
```

Reply:
Indeed, my approach was dumb. I've taken a new approach which seems to work well. I only intervene at the last moment before the data is actually written.